src/portalnet/socket.rs

```diff
@@ -44,3 +49,19 @@ fn find_assigned_ip() -> Option<IpAddr> {
     }
     None
 }
+
+#[cfg(windows)]
+fn find_assigned_ip() -> Option<IpAddr> {
+    let adapters = ipconfig::get_adapters().unwrap();
+
+    for adapter in adapters.iter() {
```

Comment:
For me, too many nested if statements can be a bit of a code smell. Especially in rust where they have quite a few handy mechanisms like `.map()` / `.filter()` / `.is_empty()` . I'd give this another go and see if you can trade out some of these nested ifs for something "flatter".

Reply:
So I tried to work out a solution using a filter, but I just couldn't figure a good one out for this here is a picture of what I tried 
![image](https://user-images.githubusercontent.com/31669092/125388692-32e0f780-e35d-11eb-98f7-ea1eea1d6a17.png)

You guys can problem come up with a better solution using a filter, but from what I could come up with using my limited knowledge of rust I think the first attempt I had at this problem was a lot cleaner, but maybe I just think that way cause I am just not as familiar with Rust.

Let me know what you think

Reply:
I would of course remove the prints by the way just there cause I was testing

Reply:
I'm also pretty new to rust, so it's tough for me to look at this and see a solution that's obviously better. Also, my compiler won't compile this code since I'm not on windows, which makes it harder since I usually like to tinker around with these things to see what does/doesn't work. This is the best I can do, but honestly, i'm not sure if it even works - but maybe it'll spark an idea on your end or maybe not. 

```rust
    ipconfig::get_adapters()
        .filter(|adapter| !adapter.gateways().is_empty())
        .map(|adapter| adapter.ip_addresses())
        .flatten()
        .find_map(|ip_addr| IpAddr::V4(_) = ip_addr)
```

Reply:
Thanks I will take a look at it again tomorrow, from just looking at it, it seems like it will work. I haven't seen flatten() before so I will read up on it

Reply:
Hopefully it works haha, yeah I hadn't seen it before either - but just came across it while browsing https://doc.rust-lang.org/std/iter/trait.Iterator.html#provided-methods hoping to stumble across something useful

Reply:
So I ended up trying to take another look at it and this is the best I could come up with using filter and such. 

![image](https://user-images.githubusercontent.com/31669092/125702273-bd50ec64-224b-442e-9b44-4f15e5b7ae9c.png)

I guess I will start looking at cl now